pkg/provider/azure_controller_common.go

```diff
@@ -337,9 +351,38 @@ func (c *controllerCommon) attachDiskBatchToNode(ctx context.Context, subscripti
 			}
 		}
 
-		err = vmset.WaitForUpdateResult(resultCtx, future, resourceGroup, "attach_disk")
+		backoffManager := wait.Backoff{
+			Duration: updateVMUpdateRetryDuration,
+			Factor:   updateVMUpdateRetryFactor,
+			Steps:    updateVMUpdateRetrySteps,
+		}
+
+	updateRetryLoop:
```

Comment:
If we restructure the loop to only cover the case where an update is required, the flow would be clearer and we would have `future != nil`/`updateRequired` dance, e.g.

```go
err = vmset.WaitForUpdateResult(...)
for err != nil && c.vmUpdateRequired(...) {
    // backoff wait
    future, err = vmset.UpdateVMAsync(...)
    if err == nil {
        err = vmset.WaitForUpdateResult(...)
    }
}
```